### ISSUE TITLE
RET501 should exempt overriding class methods

### ISSUE THREAD

**Issue by epenet** [astral-sh/ruff#12198]
If a parent class defines a return type hint for a method, then it makes sense to keep the return None explicit.

```
class BaseClass:
    def get_value(self) -> str | None:
        return "a value"

class Class(BaseClass):
    def get_value(self) -> None:
        return None
```

Based on https://github.com/home-assistant/core/pull/115031

**Comment by AlexWaygood** [comment]
This seems reasonable to me. PR welcome.

**Comment by epenet** [comment]
Note: it appears this would be severely limited by the `multifile-analysis` restrictions.
https://github.com/astral-sh/ruff/issues/7447

**Comment by MichaReiser** [comment]
Yes, this is a good point. I don't think we can implement this today without multifile-analysis

**Comment by Avasam** [comment]
With the `@override` decorator this can be done w/o multi-fine analysis.
